src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Testing/RunTestsHandler.cs

```diff
@@ -58,7 +58,7 @@ public async Task<RunTestsPartialResult[]> HandleRequestAsync(RunTestsParams req
             EnvironmentVariables = new()
             {
                 // Reset dotnet root so that vs test console can find the right runtimes.
-                { DotnetCliHelper.DotnetRootEnvVar, string.Empty },
+                { DotnetCliHelper.DotnetRootEnvVar, Environment.GetEnvironmentVariable("DOTNET_ROOT_USER") },
```

Comment:
One thing actually - we should only set this to `DOTNET_ROOT_USER` if it isn't `EMPTY` - https://github.com/dotnet/vscode-csharp/blob/main/src/lsptoolshost/dotnetRuntimeExtensionResolver.ts#L83C60-L83C65

Otherwise we'll be putting an invalid value in `DOTNET_ROOT_USER`

Reply:
mostly good - apologies for the delay (if I take too long please ping me directly, sometimes notifications get missed in the deluge)